### ISSUE TITLE
DISABLED test_backward_ddp_outside_uneven_inputs (__main__.ProcessGroupDdpUnderDistAutogradTestWithSpawn)

### ISSUE THREAD

**Issue by mrshenli** [pytorch/pytorch#45105]
https://app.circleci.com/pipelines/github/pytorch/pytorch/216783/workflows/07366a95-f646-498b-868a-51413a9879a0/jobs/7634221/steps

```
Sep 21 12:44:02 ======================================================================
Sep 21 12:44:02 ERROR [2.142s]: test_backward_ddp_outside_uneven_inputs (__main__.ProcessGroupDdpUnderDistAutogradTestWithSpawn)
Sep 21 12:44:02 ----------------------------------------------------------------------
Sep 21 12:44:02 Traceback (most recent call last):
Sep 21 12:44:02   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_distributed.py", line 243, in wrapper
Sep 21 12:44:02     self._join_processes(fn)
Sep 21 12:44:02   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_distributed.py", line 360, in _join_processes
Sep 21 12:44:02     self._check_return_codes(elapsed_time)
Sep 21 12:44:02   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_distributed.py", line 396, in _check_return_codes
Sep 21 12:44:02     raise RuntimeError(error)
Sep 21 12:44:02 RuntimeError: Processes 5 exited with error code 10
```

```
Sep 21 12:35:07   test_backward_ddp_outside_uneven_inputs (__main__.ProcessGroupDdpUnderDistAutogradTestWithSpawn) ... 2020-09-21 12:35:07,377 ddp_under_dist_autograd_test.py:342 INFO p:process 2 t:MainThread: Running the trainer #2...
Sep 21 12:35:07 2020-09-21 12:35:07,377 ddp_under_dist_autograd_test.py:328 INFO p:process 4 t:MainThread: The remote worker is running.
Sep 21 12:35:07 2020-09-21 12:35:07,377 ddp_under_dist_autograd_test.py:344 INFO p:process 2 t:MainThread: Initing trainer process group by trainer #2 with ranks [0, 1, 2, 3]
Sep 21 12:35:07 2020-09-21 12:35:07,377 ddp_under_dist_autograd_test.py:342 INFO p:process 1 t:MainThread: Running the trainer #1...
Sep 21 12:35:07 2020-09-21 12:35:07,377 ddp_under_dist_autograd_test.py:344 INFO p:process 1 t:MainThread: Initing trainer process group by trainer #1 with ranks [0, 1, 2, 3]
Sep 21 12:35:07 2020-09-21 12:35:07,377 ddp_under_dist_autograd_test.py:342 INFO p:process 0 t:MainThread: Running the trainer #0...
Sep 21 12:35:07 2020-09-21 12:35:07,378 ddp_under_dist_autograd_test.py:344 INFO p:process 0 t:MainThread: Initing trainer process group by trainer #0 with ranks [0, 1, 2, 3]
Sep 21 12:35:07 2020-09-21 12:35:07,378 ddp_under_dist_autograd_test.py:342 INFO p:process 3 t:MainThread: Running the trainer #3...
Sep 21 12:35:07 2020-09-21 12:35:07,378 ddp_under_dist_autograd_test.py:362 INFO p:process 5 t:MainThread: Running the master process...
Sep 21 12:35:07 2020-09-21 12:35:07,379 ddp_under_dist_autograd_test.py:344 INFO p:process 3 t:MainThread: Initing trainer process group by trainer #3 with ranks [0, 1, 2, 3]
Sep 21 12:35:07 2020-09-21 12:35:07,382 ddp_under_dist_autograd_test.py:353 INFO p:process 3 t:MainThread: Waiting for shutdown signal on trainer #3...
Sep 21 12:35:07 2020-09-21 12:35:07,382 ddp_under_dist_autograd_test.py:375 INFO p:process 5 t:MainThread: Created remote rrefs on master
Sep 21 12:35:07 2020-09-21 12:35:07,383 ddp_under_dist_autograd_test.py:389 INFO p:process 5 t:MainThread: Running DDP + RPC test with simulating uneven inputs across trainers.
Sep 21 12:35:07 2020-09-21 12:35:07,412 ddp_under_dist_autograd_test.py:93 INFO p:process 4 t:Dummy-1: Initing RemoteEM with 2 3
Sep 21 12:35:07 2020-09-21 12:35:07,413 ddp_under_dist_autograd_test.py:119 INFO p:process 4 t:Dummy-2: Initing RemoteNet with 5 3
Sep 21 12:35:07 2020-09-21 12:35:07,416 ddp_under_dist_autograd_test.py:353 INFO p:process 1 t:MainThread: Waiting for shutdown signal on trainer #1...
Sep 21 12:35:07 2020-09-21 12:35:07,416 ddp_under_dist_autograd_test.py:353 INFO p:process 2 t:MainThread: Waiting for shutdown signal on trainer #2...
Sep 21 12:35:07 2020-09-21 12:35:07,416 ddp_under_dist_autograd_test.py:353 INFO p:process 0 t:MainThread: Waiting for shutdown signal on trainer #0...
Sep 21 12:35:07 ERROR:root:Caught exception: 
Sep 21 12:35:07 Traceback (most recent call last):
Sep 21 12:35:07   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_distributed.py", line 246, in wrapper
Sep 21 12:35:07     fn()
Sep 21 12:35:07   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/dist_utils.py", line 70, in new_test_method
Sep 21 12:35:07     return_value = old_test_method(self, *arg, **kwargs)
Sep 21 12:35:07   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/ddp_under_dist_autograd_test.py", line 476, in test_backward_ddp_outside_uneven_inputs
Sep 21 12:35:07     self._do_test(DdpMode.OUTSIDE, simulate_uneven_inputs=True)
Sep 21 12:35:07   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/ddp_under_dist_autograd_test.py", line 455, in _do_test
Sep 21 12:35:07     self._master_process(ddp_mode, simulate_uneven_inputs)
Sep 21 12:35:07   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/ddp_under_dist_autograd_test.py", line 377, in _master_process
Sep 21 12:35:07     ddp_mode, simulate_uneven_inputs, remote_em_rref, remote_net_rref
Sep 21 12:35:07   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/ddp_under_dist_autograd_test.py", line 423, in do_test_on_master
Sep 21 12:35:07     ddp_grads, non_ddp_grads = future.wait()
Sep 21 12:35:07   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/distributed/rpc/internal.py", line 177, in _handle_exception
Sep 21 12:35:07     raise result.exception_type(result.msg)
Sep 21 12:35:07 AssertionError: On WorkerInfo(id=0, name=worker0):
Sep 21 12:35:07 AssertionError('On WorkerInfo(id=0, name=worker0):\nAssertionError(\'Default process group is not initialized\')\nTraceback (most recent call last):\n  File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/distributed/rpc/internal.py", line 164, in _run_function\n    result = python_udf.func(*python_udf.args, **python_udf.kwargs)\n  File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/ddp_under_dist_autograd_test.py", line 186, in __init__\n    if ddp_mode in (DdpMode.INSIDE, DdpMode.OUTSIDE)\n  File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/distributed/distributed_c10d.py", line 1979, in new_group\n    _check_default_pg()\n  File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/distributed/distributed_c10d.py", line 211, in _check_default_pg\n    "Default process group is not initialized"\nAssertionError: Default process group is not initialized\n')
Sep 21 12:35:07 Traceback (most recent call last):
Sep 21 12:35:07   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/distributed/rpc/internal.py", line 164, in _run_function
Sep 21 12:35:07     result = python_udf.func(*python_udf.args, **python_udf.kwargs)
Sep 21 12:35:07   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/ddp_under_dist_autograd_test.py", line 78, in _call_method
Sep 21 12:35:07     return method(rref.local_value(), *args, **kwargs)
Sep 21 12:35:07   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/distributed/rpc/internal.py", line 177, in _handle_exception
Sep 21 12:35:07     raise result.exception_type(result.msg)
Sep 21 12:35:07 AssertionError: On WorkerInfo(id=0, name=worker0):
Sep 21 12:35:07 AssertionError('Default process group is not initialized')
Sep 21 12:35:07 Traceback (most recent call last):
Sep 21 12:35:07   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/distributed/rpc/internal.py", line 164, in _run_function
Sep 21 12:35:07     result = python_udf.func(*python_udf.args, **python_udf.kwargs)
Sep 21 12:35:07   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/distributed/ddp_under_dist_autograd_test.py", line 186, in __init__
Sep 21 12:35:07     if ddp_mode in (DdpMode.INSIDE, DdpMode.OUTSIDE)
Sep 21 12:35:07   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/distributed/distributed_c10d.py", line 1979, in new_group
Sep 21 12:35:07     _check_default_pg()
Sep 21 12:35:07   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/distributed/distributed_c10d.py", line 211, in _check_default_pg
Sep 21 12:35:07     "Default process group is not initialized"
Sep 21 12:35:07 AssertionError: Default process group is not initialized
```

cc @ezyang @gchanan @zou3519 @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @xush6528 @osalpekar @jiayisuse @agolynski @jjlilley

**Comment by rohan-varma** [comment]
@mrshenli Is this one expected to be fixed by https://github.com/pytorch/pytorch/pull/45088?

**Comment by mrshenli** [comment]
@rohan-varma this should be fixed by #45181 when that lands I think

**Comment by mrshenli** [comment]
closed by #45181
